l10n_it_pec_messages/model/mail_thread.py

```diff
@@ -85,11 +86,40 @@ def parse_daticert(self, cr, uid, daticert, context=None):
                         msg_dict['recipient_addr'] = child2.text
         return msg_dict
 
-    def get_pec_attachments(self, cr, uid, message, context=None):
-        postacert = False
-        daticert = False
-        smime = False
-        for part in message.walk():
+    def _get_msg_anomalia(self, msg):
+        to = None
+        msg_id = None
+        parser = email.Parser.HeaderParser()
+        msg_val = email.message_from_string(
+            parser.parsestr(msg.as_string()).get_payload()
+            )
+        if 'To'in msg_val:
+            to = msg_val['To']
+        if 'X-Riferimento-Message-ID'in msg_val:
+            msg_id = msg_val['X-Riferimento-Message-ID']
+        return to, msg_id
+
+    def _get_msg_delivery(self, msg):
+        for dsn in msg.get_payload():
+            if 'Action' in dsn:
+                return dsn['Action']
+
+    def _get_msg_payload(self, cr, uid, msg, parts={}, num=0):
+        """
+        
+	This method recursively checks the message structure
+        and saves the informations (bodies, attachments,
+        pkcs7 signatures, etc.) in a dictionary.
+        
+	The parameters:
+        
+	num msg is the multipart message to process
+        parts is the dictionary where the informations are saved
+        num is an integer that refers to the deepness
+        of the content in the original mail
```

Comment:
Scusa non capisco questa parte: cos'è la mail originale? In che relazione è con msg? Cosa si intende con deepness?